tests/periph_timer/main.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Peripheral timer test application
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "periph/timer.h"
+
+/**
+ * @brief   Make sure, the maximum number of timers is defined
+ */
+#ifndef TIMER_NUMOF
+#error "TIMER_NUMOF not defined!"
+#endif
+
+#define MAX_CHANNELS        (10U)
+#define TIM_SPEED           (1U)        /* try to run with 1MHz */
+#define CHAN_OFFSET         (5000U)     /* fire every 5ms */
+
+static volatile int fired;
+static volatile uint32_t sw_count;
+static volatile uint32_t timeouts[MAX_CHANNELS];
+
+static void cb(int chan)
+{
+    timeouts[chan] = sw_count;
+    fired++;
+}
+
+static int test_timer(int num)
```

Comment:
Nope, you are right. Somehow I was convinced, that the loop in line 124 was using and `int` as counter variable.... Sorry for this discussion - clearly my fault...